node/utils/setFilterVisibility.ts

```diff
@@ -0,0 +1,112 @@
+import { VBase } from '@vtex/api'
+import { clone, flatten } from 'ramda'
+import { Search } from '../clients/search'
+import {
+  LIST_SPECIFICATIONS_BY_CATEGORY_ID,
+  SPECIFICATION_BUCKET,
+} from '../resolvers/search/constants'
+import { Attribute, TextAttribute } from './attributes'
+import { staleFromVBaseWhileRevalidate } from './vbase'
+
+const getSpecificationsByCategoryId = (
+  vbase: VBase,
+  search: Search,
+  categoryId: number
+) => {
+  return staleFromVBaseWhileRevalidate(
+    vbase,
+    `${LIST_SPECIFICATIONS_BY_CATEGORY_ID}`,
+    categoryId.toString(),
+    async (params: { categoryId: number; search: Search }) =>
+      params.search.specificationsByCategoryId(params.categoryId),
+    { categoryId, search },
+    {
+      expirationInMinutes: 10,
+    }
+  )
+}
+
+const isFilterByFieldId = async (
+  vbase: VBase,
+  search: Search,
+  fieldId: number
+) => {
+  const specification = await staleFromVBaseWhileRevalidate(
+    vbase,
+    `${SPECIFICATION_BUCKET}`,
+    fieldId.toString(),
+    async (params: { fieldId: number; search: Search }) =>
+      params.search.getField(params.fieldId),
+    { fieldId, search },
+    {
+      expirationInMinutes: 10,
+    }
+  )
+
+  return specification.IsFilter
+}
+
+const setFilterVisibility = async (
+  vbase: VBase,
+  search: Search,
+  attributes: Attribute[]
+) => {
+  const hiddenSpecificationsMap = new Map<string, boolean>()
+  const categoryRegex = /^category-[0-9]+$/
+
+  const categoryValues = flatten(
```

Comment:
is there any way to avoid using a `ramda` function here?

Reply:
I could create those functions by myself. but since it's a back end app, I don't see any problem in importing them